packages/headless/src/features/search-parameters/search-parameter-selectors.ts

```diff
@@ -19,7 +19,9 @@ export function initialSearchParameterSelector(
       state.advancedSearchQueries?.defaultFilters.cq ??
       getAdvancedSearchQueriesInitialState().defaultFilters.cq,
     firstResult: getPaginationInitialState().firstResult,
-    numberOfResults: getPaginationInitialState().numberOfResults,
+    numberOfResults:
+      state.pagination?.defaultNumberOfResults ??
+      getPaginationInitialState().numberOfResults,
```

Comment:
how about `getPaginationInitialState().defaultNumberOfResults` to make it a bit clearer?

Reply:
* clearer that you are actually just talking about default values here, and nothing that could change